python/core/auto_generated/qgsabstractdatabaseproviderconnection.sip.in

```diff
@@ -359,6 +359,40 @@ Raises a QgsProviderConnectionException if any errors are encountered.
 :raises :: py:class:`QgsProviderConnectionException`
 %End
 
+    virtual void deleteField( const QString &fieldName, const QString &schema, const QString &tableName, bool force = false ) const throw( QgsProviderConnectionException );
+%Docstring
+Deletes the field with the specified name.
+Raises a QgsProviderConnectionException if any errors are encountered.
+
+:param fieldName: name of the field to be deleted
+:param schema: name of the schema (schema is ignored if not supported by the backend).
+:param tableName: name of the table
+:param force: if ``True``, a DROP CASCADE will drop all related objects
```

Comment:
I'm thinking this should be an extra capability flag, e.g. DeleteFieldCascade. Then we could expose a checkbox for "cascade" delete for compatible providers.